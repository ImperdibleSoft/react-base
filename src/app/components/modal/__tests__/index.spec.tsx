import * as React from 'react';
import { shallow } from 'enzyme';

import Modal from '..';

describe('Modal', () => {
  const modalRoot = document.createElement('div');
  modalRoot.setAttribute('id', 'modal-root');
  document.body.appendChild(modalRoot);

  it('should render a closed modal', () => {
    const component = shallow(
      <Modal handleClose={() => {}} isOpen={false}>
        <p>Test children</p>
      </Modal>
    );

    expect(component).toMatchSnapshot();
  });

  it('should render an open modal', () => {
    const component = shallow(
      <Modal handleClose={() => {}} isOpen={true}>
        <p>Test children</p>
      </Modal>
    );

    expect(component).toMatchSnapshot();
  });
});
