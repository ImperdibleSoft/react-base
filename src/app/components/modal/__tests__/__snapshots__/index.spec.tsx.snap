// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal should render a closed modal 1`] = `""`;

exports[`Modal should render an open modal 1`] = `
<Portal
  containerInfo={
    <div
      id="modal-root"
    />
  }
>
  <GlobalStyleComponent />
  <Backdrop
    onClick={[Function]}
  />
  <Wrapper>
    <p>
      Test children
    </p>
  </Wrapper>
</Portal>
`;
